### DISCUSSION TITLE
[Art competition] Program synthesis

### DISCUSSION THREAD

**Discussion by KunJeong** [prosyslab-classroom/cs424-program-reasoning#253] [Art competition]
![DALL·E 2022-11-30 16 06 23 - a conveyor belt with clothes in a clothing factory, and at the end of the conveyor is a large fire](https://user-images.githubusercontent.com/10477004/204730650-b1406ecb-f86f-46d9-a6b9-9dc77a0592bc.png)

## Short Description
For me, [Program synthesis] means [a conveyor belt with clothes in a clothing factory, and at the end of the conveyor is a large fire]

## Long Description
In program synthesis, we are essentially searching for “clothes that fit.” To do this, we are required to 1. Generate probable programs (i.e. similar-looking clothes) quickly and efficiently, and 2. Prune candidates as much as possible before trying them on. I imagined a clothing factory where similar but varying clothes are made quickly, and then most of them burned so that only some may survive to be tried on.

**Comment by KihongHeo** [comment]
Why burn the clothes?

**Comment by KunJeong** [comment]
Because for most of them, we can know whether they fit or not before trying them on! For instance, we can eliminate all pants that are too short, too long, have a too small or wide waist, etc (in a sense, this is a bit similar to top-down propagation). We can also eliminate sets of pants that are more or less the same in terms of fit (=equivalence reduction). So we remove them they are tried on. I thought "burning" because we are removing enormous piles of clothes at a time.
